frontend/src/metabase/containers/formUtils.js

```diff
@@ -100,3 +100,12 @@ export function cleanObject(object) {
   });
   return result;
 }
+
+export function isNestedFieldName(name) {
+  return name.includes(".");
+}
+
+export function getMaybeNestedValue(obj, fieldName) {
+  const isNestedField = fieldName.includes(".");
+  return isNestedField ? getIn(obj, fieldName.split(".")) : obj[fieldName];
```

Comment:
could DRY this up a bit
```suggestion
  return isNestedFieldName(fieldName) ? getIn(obj, fieldName.split(".")) : obj[fieldName];
```